Add vaultwarden (ex. bitwarden_rs)

hgy59 opened the pull request:
## Description

<!--Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.-->
This PR is based on the work of @Hylen for a bitwarden_rs package (see https://github.com/Hylen/spksrc/tree/bitwarden).
Meanwhile the name has changed from bitwarden_rs to vaultwarden.

I got the main components working with the latest version 1.25.0 (released two days ago).
Still some work todo for MySQL/MariaDB integration and package updates.

Fixes #3712
Fixes #6916

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relavent tags.-->
- [x] New Package
- [x] This change requires a documentation update (e.g. Wiki)

Hylen committed:
First commit.

Hylen committed:
More rust improvements.

Hylen committed:
Added libmysqlclient.

Hylen committed:
Revert "Added libmysqlclient."

This reverts commit 0dfd636a5c9b2a55329fb9ec002c8cf04ce8a4ab.

Hylen committed:
Add bitwarden_rs spk makefile.

Hylen committed:
Fix the plist.

hgy59 committed:
vaultwarden: update from bitwarden_rs

- udpate to vaultwarden v1.25.0 and vaultwarden_web v2.28.1
- initial service setup
- initial installation wizard

hgy59 committed:
mysql-connector-c: include libatomic from toolchain

- include libatomic from toolchain for newer glic (i.e. DSM >= 7)

hgy59 committed:
mariadb-connector-c: remove obsolete entry from PLIST

hgy59 committed:
use framework to include libatomic from toolchain

hgy59 committed:
vaultwarden add support for postgresql

- add libpq (postgresql client library)
- add and document unsupported archs

hgy59 committed:
vaultwarden: update configuration

- activate .env file
- adjust config data

hgy59 committed:
adjust configuration

- use valid json file to pass json lint checks

hgy59 committed:
update libpq to v11.17

hgy59 committed:
udpate vaultwarden to v1.25.2

hgy59 committed:
update vaultwarden_web to v2022.9.0

hgy59 committed:
adjust for changes in PR #5466

hgy59 committed:
update vaultwarden

- update vaultwarden to v1.27.0
- update vaultwarden_web to v2023.1.1

hgy59 committed:
remove installation of patchelf

hgy59 commented on mk/spksrc.cross-rust.mk:
```diff
@@ -56,6 +56,7 @@ CARGO_INSTALL_ARGS += --root $(STAGING_INSTALL_PREFIX)
 # Default build with rust and installation with cargo
 rust_install_target:
 	@echo "  ==> Cargo install rust package $(PKG_NAME)"
+	@$(RUN) rustup target install $(RUST_TARGET)
```
@th0ma7 without this change, only x64 target builds do not fail. (local and github builds).
is there a better solution for this?

th0ma7 replied:
This is odd... Would you hapen to be running on an "unclean" tree?  I just tried it from a clean tree within `cross` and it worked out ok without any need of patching.

Try cleaning-up your toolchain directory or simply delete the rust status file from there.  Normally that should do it.

hgy59 replied:
> This is odd... Would you hapen to be running on an "unclean" tree? I just tried it from a clean tree within `cross` and it worked out ok without any need of patching.
> 
> Try cleaning-up your toolchain directory or simply delete the rust status file from there. Normally that should do it.

It failed also on github build action that does the delete of rust status now.

th0ma7 replied:
This is really really weird.  Yesterday I added `dssim` to `gstreamer` PR https://github.com/SynoCommunity/spksrc/pull/5298 which is rust based (`spksrc.cross-rust.mk`).  I was expecting the PR to fail similarly to yours but it didn't.  Everything worked out just fine.

Could it be possible that your PR's build image file is polluted somehow?  Could a rebase to master solve things?

hgy59 replied:
@th0ma7 just rebased this PR and removed the line above. Still the same. Only x64 package build without the additional rustup target install ... line.
So I have added it again.

th0ma7 replied:
I may have found hints of what`s going on, using a clean image and tree I was able to reproduce your issue.

When digging further I noticed the following:
```
spksrc@spksrc:~/hgy59$ find . -name cargo | grep bin
./spksrc/distrib/cargo/bin/cargo
./spksrc/distrib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo
./spksrc/distrib/rustup/toolchains/1.66.0-x86_64-unknown-linux-gnu/bin/cargo

spksrc@spksrc:~/hgy59$ ll ./spksrc/distrib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo ./spksrc/distrib/rustup/toolchains/1.66.0-x86_64-unknown-linux-gnu/bin/cargo
-rwxr-xr-x 1 spksrc spksrc 25722120 Feb 25 16:07 ./spksrc/distrib/rustup/toolchains/1.66.0-x86_64-unknown-linux-gnu/bin/cargo
-rwxr-xr-x 1 spksrc spksrc 26054568 Feb 25 15:55 ./spksrc/distrib/rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo
```
For some odd reason `cargo` switches from "stable" install at toolchain time to an actual version being `1.66.0` at build time ...  still investigating... but getting closer.

th0ma7 replied:
The following looks like solving the version changing issue, can you test that out?
```
$ git diff .
diff --git a/mk/spksrc.cross-rust-env.mk b/mk/spksrc.cross-rust-env.mk
index f398432e..a3a7d14c 100644
--- a/mk/spksrc.cross-rust-env.mk
+++ b/mk/spksrc.cross-rust-env.mk
@@ -8,7 +8,9 @@ export RUSTUP_HOME=$(BASE_DISTRIB_DIR)/rustup
 export PATH:=$(BASE_DISTRIB_DIR)/cargo/bin:$(PATH)
 endif
 
-RUST_TOOLCHAIN ?= stable
+ifeq ($(RUST_TOOLCHAIN),)
+RUST_TOOLCHAIN = stable
+endif
 
 RUST_TARGET =
 # map archs to rust targets
diff --git a/mk/spksrc.cross-rust.mk b/mk/spksrc.cross-rust.mk
index 534f708c..d76fa1bc 100644
--- a/mk/spksrc.cross-rust.mk
+++ b/mk/spksrc.cross-rust.mk
@@ -56,8 +56,7 @@ CARGO_INSTALL_ARGS += --root $(STAGING_INSTALL_PREFIX)
 # Default build with rust and installation with cargo
 rust_install_target:
        @echo "  ==> Cargo install rust package $(PKG_NAME)"
-       @$(RUN) rustup target install $(RUST_TARGET)
-       @$(RUN) cargo install $(CARGO_INSTALL_ARGS)
+       @$(RUN) cargo +$(RUST_TOOLCHAIN) install $(CARGO_INSTALL_ARGS)
 
 
```

hgy59 committed:
update cross/vaultwarden/digests

hgy59 committed:
update vaultwarden

- update vaultwarden to v1.28.1
- update vaultwarden_web to v2023.3.0b
- update native-cmake-env to use additional c/cxx flags

hgy59 committed:
fix for DSM < 7

hgy59 committed:
fix build for comcerto2k for DSM>=7

hgy59 committed:
ARMv7L_ARCHS are not supported, update vaultwarden_web to v2023.4.2

hgy59 committed:
libpq: update to v11.22 and use cross/openssl3

hgy59 committed:
update vaultwarden

- update vaultwarden to v1.30.1
- update vaultwarden_web to v2024.1.0
- use cross/openssl3

mreid-tt commented:
hey @hgy59, would you like me to help bring this to completion?

mreid-tt committed:
vaultwarden: Update to v1.35.2 and web vault v2025.12.1

mreid-tt committed:
vaultwarden: Replace mysql-connector-c with mariadb-connector-c

mysql-connector-c is deprecated, mariadb-connector-c is the actively
maintained replacement and is already used by other packages.

mreid-tt committed:
vaultwarden: Disable web UI shortcut and add HTTPS warning

Vaultwarden requires HTTPS for the Web Crypto API. Since the package
cannot automatically configure a reverse proxy, the web UI shortcut
is disabled via NO_SERVICE_SHORTCUT to prevent users from seeing an
error page. The SERVICE_PORT is retained for firewall configuration.

Users must configure a reverse proxy with SSL before accessing the
web interface. The install wizard now includes a prominent warning
and link to the wiki for setup instructions.

mreid-tt committed:
vaultwarden: Only run postinst setup on fresh install

Wrap service_postinst logic in SYNOPKG_PKG_STATUS check to skip
config file creation during upgrades. This preserves user settings
and avoids running wizard-dependent code when wizard values are
not available.

Also add error logging if env or config file creation fails.

mreid-tt committed:
Merge branch 'master' into pr/5284

mreid-tt committed:
Revert unrelated cmake framework change

Remove CMAKE_C_FLAGS/CMAKE_CXX_FLAGS additions to native-cmake-env.mk

mreid-tt committed:
vaultwarden: Fix mysqlclient-sys build for cross-compilation

Add environment variables to help mysqlclient-sys Rust crate find the
MariaDB client library when cross-compiling:
- MYSQLCLIENT_LIB_DIR and MYSQLCLIENT_INCLUDE_DIR
- PQ_LIB_DIR for PostgreSQL client
- PATH to include mysql_config script
- MYSQLCLIENT_VERSION based on toolchain GCC version

mreid-tt committed:
vaultwarden: Fix display name capitalization

mreid-tt committed:
libpq: Update to v16.11
